alica_engine/include/engine/planselector/PartialAssignment.h

```diff
@@ -52,30 +52,27 @@ class PartialAssignment final : public IAssignment
     std::shared_ptr<std::list<const supplementary::AgentID*>> getRobotsWorkingAndFinished(const EntryPoint* ep) override;
     std::shared_ptr<std::list<const supplementary::AgentID*>> getRobotsWorkingAndFinished(int64_t epid) override;
     std::shared_ptr<std::list<const supplementary::AgentID*>> getUniqueRobotsWorkingAndFinished(const EntryPoint* ep) override;
-    bool addIfAlreadyAssigned(shared_ptr<SimplePlanTree> spt, const supplementary::AgentID* robot);
+    bool addIfAlreadyAssigned(std::shared_ptr<SimplePlanTree> spt, const supplementary::AgentID* robot);
     bool assignRobot(const supplementary::AgentID* robotId, int index);
-    shared_ptr<list<PartialAssignment*>> expand();
+    std::shared_ptr<std::list<PartialAssignment*>> expand();
     bool isValid() const override;
     bool isGoal();
     static bool compareTo(PartialAssignment* thisPa, PartialAssignment* newPa);
-    std::string toString();
-    AssignmentCollection* getEpRobotsMapping();
+    std::string toString() const override;
+    virtual AssignmentCollection* getEpRobotsMapping() const override { return epRobotsMapping; }
     const Plan* getPlan() const { return plan; }
     std::shared_ptr<UtilityFunction> getUtilFunc();
     std::shared_ptr<SuccessCollection> getEpSuccessMapping();
-    std::string assignmentCollectionToString();
+    std::string assignmentCollectionToString() const override;
     std::shared_ptr<std::vector<EntryPoint*>> getEntryPoints();
```

Comment:
no longer exists, you can get rid of this

Reply:
nice find